### PR TITLE
1301: Porting unit tests to repo

### PR DESCRIPTION
**Pull request by avgAGB**

### CODE REVIEW

**Review by jwomeara — DISMISSED** [review]
LGTM

**Review by apmoriarty — APPROVED** [review]

File: warehouse/query-core/src/test/java/datawave/query/rewrite/jexl/visitors/PushdownLargeFieldedListsVisitorTest.java [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+        conf.setMaxOrExpansionFstThreshold(100);
+        conf.setIndexedFields(Sets.newHashSet("FOO", "BAR", "FOOBAR"));
+    }
+    
+    @Test
+    public void testSimpleExpression() throws Throwable {
+        String rewritten = JexlStringBuildingVisitor.buildQuery(PushdownLargeFieldedListsVisitor.pushdown(conf,
+                        TreeFlatteningRebuildingVisitor.flatten(JexlASTHelper.parseJexlQuery("FOO == 'BAR'")), null, null));
+        Assert.assertEquals("FOO == 'BAR'", rewritten);
+    }
```
Comment (apmoriarty) [inline warehouse/query-core/src/test/java/datawave/query/rewrite/jexl/visitors/PushdownLargeFieldedListsVisitorTest.java]: For tests like these I would like to see something like this, basically extract common code to a test method. It seems like overkill for the case of a single equality node, but as the query gets more complex the tests become easier to read.
```
public void testSimpleExpression(){
  String query = "FOO == 'bar'";
  test(query, query);
}
...
private void test(String query, String expected){
  test(query, expected, config);  // pass in the default shard query config
}

private void test(String query, String expected, ShardQueryConfiguration config){
  ASTJexlScript script = JexlASTHelper.parseJexlQuery(query);
  ASTJexlScrit pushed = PushdownLargeFieldedLists(confg, script, null, null)
}
```

File: warehouse/query-core/src/test/java/datawave/query/rewrite/jexl/visitors/PushdownLargeFieldedListsVisitorTest.java [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+    
+    @Test
+    public void testMultipleExpression() throws Throwable {
+        String rewritten = JexlStringBuildingVisitor.buildQuery(PushdownLargeFieldedListsVisitor.pushdown(conf,
+                        TreeFlatteningRebuildingVisitor.flatten(JexlASTHelper.parseJexlQuery("FOO == 'BAR' || FOO == 'FOO' || BAR == 'FOO'")), null, null));
+        String expected = "BAR == 'FOO' || FOO == 'BAR' || FOO == 'FOO'";
+        Assert.assertEquals("EXPECTED: " + expected + "\nACTUAL: " + rewritten, expected, rewritten);
```
Comment (apmoriarty) [inline warehouse/query-core/src/test/java/datawave/query/rewrite/jexl/visitors/PushdownLargeFieldedListsVisitorTest.java]: my personal preference is static imports for asserts, this isn't a blocker

File: warehouse/query-core/src/test/java/datawave/query/rewrite/jexl/visitors/PushdownLargeFieldedListsVisitorTest.java [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+                        rewritten);
+    }
+    
+    @Test
+    public void testEscapedValues() throws Throwable {
+        String rewritten = JexlStringBuildingVisitor.buildQuery(PushdownLargeFieldedListsVisitor.pushdown(conf,
+                        TreeFlatteningRebuildingVisitor.flatten(JexlASTHelper.parseJexlQuery("FOO == 'BAR' || FOO == 'FOO' || FOO == 'FOO,BAR'")), null, null));
+        String id = rewritten.substring(rewritten.indexOf("id = '") + 6, rewritten.indexOf("') && (field"));
+        Assert.assertEquals("((_List_ = true) && ((id = '" + id + "') && (field = 'FOO') && (params = '{\"values\":[\"BAR\",\"FOO\",\"FOO,BAR\"]}')))",
+                        rewritten);
```
Comment (apmoriarty) [inline warehouse/query-core/src/test/java/datawave/query/rewrite/jexl/visitors/PushdownLargeFieldedListsVisitorTest.java]: You can still extract common code in the case of a pushed visitor. You could do something like this
```
public void test(){
  String query = 'your query here"
  String expectedLeft = "((_List_ = true) && ((id = '";
  String expectedRight = "the rest of your quey here";
  test(query, expectedLeft, expectedRight, config);
}
...
// and then essentially a copy of the original test method, but with updated asserts to
// handle the pushed value query
```

File: warehouse/query-core/src/test/java/datawave/query/rewrite/jexl/visitors/PushdownLargeFieldedListsVisitorTest.java [hunk 4]
```diff
@@ -0,0 +1,186 @@
...
+                                        null, null));
+        String id1 = rewritten.substring(rewritten.indexOf("id = '") + 6, rewritten.indexOf("') && (field"));
+        String id2 = rewritten.substring(rewritten.lastIndexOf("id = '") + 6, rewritten.lastIndexOf("') && (field"));
+        Assert.assertEquals("f:includeRegex(FOO, 'blabla') && X == 'Y' && (((_List_ = true) && ((id = '" + id1
+                        + "') && (field = 'BAR') && (params = '{\"values\":[\"BAR\",\"FOO\",\"FOOBAR\"]}'))) || ((_List_ = true) && ((id = '" + id2
+                        + "') && (field = 'FOO') && (params = '{\"values\":[\"BAR\",\"FOO\",\"FOOBAR\"]}')))) && !(Y == 'X')", rewritten);
+    }
+    
+    @Test
+    public void testPushdownFst() throws Throwable {
```
Comment (apmoriarty) [inline warehouse/query-core/src/test/java/datawave/query/rewrite/jexl/visitors/PushdownLargeFieldedListsVisitorTest.java:98]: Are there plans to add more comprehensive FST test? If so, might be worth extracting some of these variables/functions

**Review by apmoriarty — CHANGES_REQUESTED** [review]

File: warehouse/query-core/src/test/java/datawave/query/rewrite/jexl/visitors/PushdownLargeFieldedListsVisitorTest.java (hunk 1 above)
Comment (avgAGB) [reply warehouse/query-core/src/test/java/datawave/query/rewrite/jexl/visitors/PushdownLargeFieldedListsVisitorTest.java]: I understand the reformatting suggestions, though my question is would that cause an issue if the unit test code here looks different than it does elsewhere? The acceptance criteria on the ticket seemed to suggest the unit test code be moved here as it existed before.

EDIT: Per our sidechat, I'll made adjustments in future commits and look to remove redundant copies once this is merged.

File: warehouse/query-core/src/test/java/datawave/query/rewrite/jexl/visitors/PushdownLargeFieldedListsVisitorTest.java (hunk 2 above)
Comment (avgAGB) [reply warehouse/query-core/src/test/java/datawave/query/rewrite/jexl/visitors/PushdownLargeFieldedListsVisitorTest.java]: Will adjust this in the next commit

File: warehouse/query-core/src/test/java/datawave/query/rewrite/jexl/visitors/PushdownLargeFieldedListsVisitorTest.java (hunk 3 above)
Comment (avgAGB) [reply warehouse/query-core/src/test/java/datawave/query/rewrite/jexl/visitors/PushdownLargeFieldedListsVisitorTest.java]: Updated in latest commit.

File: warehouse/query-core/src/test/java/datawave/query/rewrite/jexl/visitors/PushdownLargeFieldedListsVisitorTest.java (hunk 4 above)
Comment (avgAGB) [reply warehouse/query-core/src/test/java/datawave/query/rewrite/jexl/visitors/PushdownLargeFieldedListsVisitorTest.java:98]: I don't have plans to add a more comprehensive test for this particular ticket, however I tried to pull out some of the setup and the pushdown to keep it in line with the other unit tests.

**Review by ivakegg — APPROVED** [review]

**Review by apmoriarty — APPROVED** [review]

**Pull request merged**